feat: frontend voting price curves

nakedfool opened the pull request:
TODO:

- [x] Add flat/exponential model on the contest page (price updates on the contest page and the overall logic for it so it's reusable on other parts of the site)
- [x] Add flat/exponential model to user store, we need to refactor the store since now we have dynamic price of the vote, current code is now stale since before we had a model where cost to vote was always static (user vote amount should change per `PRICE_CURVE_UPDATE_INTERVAL` and therefore, user vote amount needs to reflect those changes)
- [x] Update dynamic vote price on voting modal and in the entry page
- [x] Update rules tab to include dynamic vote price info

divine-economy commented:
looking great

two quick notes @nakedfool when you're back:

1. we don't need the space between the number and ETH -- we can go coinbase style where it's just 0.0001ETH rather than 0.0001 ETH, and the difference is demarcated by the color and lack of line

2. we should think about what the default bottom price is @siobh9. right now it's about 2 cents, which could work. i was imagining we get it down to 1 cent so we could get a 1000x multiple to make things as dramatic as possible to incentivize early participation, but am open here: would just rather start with a default multiple that's too high than too low (so that everyone decides it's worth waiting until the end), and then can always adjust this over time.

nakedfool commented:
> * we don't need the space between the number and ETH -- we can go coinbase style where it's just 0.0001ETH rather than 0.0001 ETH, and the difference is demarcated by the color and lack of line
> * we should think about what the default bottom price is @siobh9. right now it's about 2 cents, which could work. i was imagining we get it down to 1 cent so we could get a 1000x multiple to make things as dramatic as possible to incentivize early participation, but am open here: would just rather start with a default multiple that's too high than too low (so that everyone decides it's worth waiting until the end), and then can always adjust this over time.

1. Fixed, chain native currency label should be next to the input now.

2. Let's keep our default cost to vote for now for minimum price and i adjusted end price to be x1000 as default end price.

Build is failing right now, but as soon as it's stable, you will be able to see these changes.

siobh9 reviewed (approved):
lgtm!
